src/JSURI.js

```diff
@@ -36,6 +36,22 @@ function _decodeURIComponent(fail, succeed, input) {
 
 exports._decodeURIComponent = _decodeURIComponent;
 
-exports._decodeFormURLComponent = function encode(fail, succeed, input) {
+exports._decodeFormURLComponent = function _decodeFormURLComponent(fail, succeed, input) {
   return _decodeURIComponent(fail, succeed, input.replace(/\+/g, " "));
 };
+
+exports._encodeURI = function _encodeURI(fail, succeed, input) {
+  try {
+    return succeed(encodeURI(input));
```

Comment:
We should follow RFC3896, as was done for `encodeURIComponent` above -- though the function to fix the encoded string is a bit different. See here for the implementation:

https://web.archive.org/web/20210117175449/https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI#encoding_for_ipv6

Could you please apply this helper function to `encodeURI` here, and include the archive.org link as was done for the `toRFC3986` function above? Since this helper isn't being reused anywhere I don't think it needs to be its own helper in the file; I think it can just be applied as was done in the MDN documentation directly to `encodeURI`.

Reply:
:+1: